src/utils/payloadStringify.js

```diff
@@ -92,6 +95,61 @@ const getItemsInPath = safeExecute(
   []
 );
 
+function scrubJsonStringBySecretPath(input, requestedPath, currentPath) {
+  try {
+    let realJson = JSON.parse(input);
+    const res = innerPathScrubbing(realJson, requestedPath, currentPath);
+    return JSON.stringify(res);
+  } catch (e) {
+    logger.debug('Failed to parse json payload for path scrubbing', { error: e, event: input });
+    return innerPathScrubbing(input, requestedPath, currentPath);
+  }
+}
+
+function scrubJsonBySecretPath(input, secretPaths, currentPath) {
+  return innerPathScrubbing(input, secretPaths, currentPath);
+}
+
+function keyExistsInPaths(paths, key) {
+  let allPathKeys = [];
+  paths.forEach((path) => {
+    let keys = path.split('.');
+    allPathKeys = [...allPathKeys, ...keys];
+  });
+  const uniquePaths = allPathKeys.filter((x, i) => i === allPathKeys.indexOf(x));
+  logger.debug(`Checking if key ${key} exists in ${uniquePaths}`);
```

Comment:
i'm not sure if it's worth it, but seems like the creation of `uniquePaths` array can happen just once, instead of each iteration of `innerPathScrubbing`